modules/reindex/build.gradle

```diff
@@ -16,24 +16,20 @@ import org.elasticsearch.gradle.transform.UnzipTransform
 
 apply plugin: 'elasticsearch.test-with-dependencies'
 apply plugin: 'elasticsearch.jdk-download'
-apply plugin: 'elasticsearch.legacy-yaml-rest-test'
-apply plugin: 'elasticsearch.legacy-java-rest-test'
-apply plugin: 'elasticsearch.legacy-yaml-rest-compat-test'
+apply plugin: 'elasticsearch.internal-yaml-rest-test'
+apply plugin: 'elasticsearch.internal-java-rest-test'
+apply plugin: 'elasticsearch.yaml-rest-compat-test'
 apply plugin: 'elasticsearch.internal-cluster-test'
 
 esplugin {
   description 'The Reindex module adds APIs to reindex from one index to another or update documents in place.'
   classname 'org.elasticsearch.reindex.ReindexPlugin'
 }
 
-testClusters.configureEach {
-  // Modules who's integration is explicitly tested in integration tests
-  module ':modules:parent-join'
-  module ':modules:lang-painless'
-  module ':modules:rest-root'
-  // Whitelist reindexing from the local node so we can test reindex-from-remote.
-  setting 'reindex.remote.whitelist', '127.0.0.1:*'
-  requiresFeature 'es.index_mode_feature_flag_registered', Version.fromString("8.0.0")
+if (buildParams.isSnapshotBuild() == false) {
```

Comment:
Is this bit required? It wasn't in the old build script. Was this perhaps just copy/pasted from elsewhere?

Reply:
This is my attempt to replicate `index_mode_feature_flag_registered` that you asked about below. I realize now that it is a noop since it does not actually apply to a cluster. I have no clue what `index_mode_feature_flag_registered` does but it seems to not impact anything indeed. I'll research a bit.

Reply:
You are correct, you need to add the feature flag to the cluster. Feature flags are only required when running the tests in "release" mode. Maybe this flag isn't required anymore and is enabled by default in current versions so indeed may be unnecessary now. Probably worth confirming.

Reply:
Last usage of `index_mode_feature_flag_registered` was removed in #95220. It's a complete noop and should be removed.

Reply:
Awesome. This can be done separately in a follow up.